monitors-driver-library/src/main/java/cloud/benchflow/monitors/MonitorFactory.java

```diff
@@ -0,0 +1,69 @@
+package cloud.benchflow.monitors;
+
+import cloud.benchflow.monitors.cpu.CpuMonitor;
+import cloud.benchflow.monitors.querymysql.QueryMySQLMonitor;
+
+import com.sun.faban.driver.transport.hc3.ApacheHC3Transport;
+
+import java.util.Map;
+import java.util.logging.Logger;
+
+/**
+ * @author Simone D'Avico (simonedavico@gmail.com)
+ *
+ * Created on 13/07/16.
+ */
+public class MonitorFactory {
+
+    public static Monitor getMonitor(ApacheHC3Transport http,
+                                     String monitorName,
+                                     Map<String, String> parameters,
+                                     String endpoint,
+                                     String startApi,
+                                     String stopApi,
+                                     String monitorApi) {
+
+        MonitorAPI api = new MonitorAPI(startApi, monitorApi, stopApi);
+        switch (monitorName) {
```

Comment:
After thinking a little more about this, we wouldn't get any advantage from this, only code that uses reflection and is more difficult to understand.

This is because to select what instance of Monitor I should return I would need to have a mapping somewhere between monitor name (what is in the run.xml file) and the corresponding Monitor interface implementation.

This would basically be the same as branching on monitor name, exactly as I'm doing now without reflection.